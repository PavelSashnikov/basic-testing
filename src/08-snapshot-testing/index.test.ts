import { generateLinkedList } from './index';

describe('generateLinkedList', () => {
  const arr = ['a', 'b', 'c', 'd'];
  const linkedList = {
    value: 'a',
    next: {
      value: 'b',
      next: {
        value: 'c',
        next: { value: 'd', next: { value: null, next: null } },
      },
    },
  };
  // Check match by expect(...).toStrictEqual(...)
  test('should generate linked list from values 1', () => {
    const list = generateLinkedList(arr);
    expect(list).toStrictEqual(linkedList);
  });

  // Check match by comparison with snapshot
  test('should generate linked list from values 2', () => {
    const list = generateLinkedList(arr);
    expect(list).toMatchSnapshot();
  });
});
