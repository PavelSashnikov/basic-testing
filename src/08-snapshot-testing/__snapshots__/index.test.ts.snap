// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generateLinkedList should generate linked list from values 2 1`] = `
Object {
  "next": Object {
    "next": Object {
      "next": Object {
        "next": Object {
          "next": null,
          "value": null,
        },
        "value": "d",
      },
      "value": "c",
    },
    "value": "b",
  },
  "value": "a",
}
`;
